### PR TITLE
gives event-spawned xeno larva  a growth boost

### PR DESCRIPTION
**Pull request by TDSSS**
**What does this PR do:**
TLDR: the larvas spawned by a xeno event now start 3/4 grown up already, so they can evolve without as much needless waiting around.

We all know how the typical Xeno event goes, somewhat like this:

1. event fires and two larva are spawned in. They hide, hopefully crew doesn't spot them already at this point.
2. Wait for larva to mature, nothing really happens in this stage, larvas look for turbine or similar abandoned location.
3. larvas evolve, one hopefully goes drone to become queen. Hunter can go out and start capturing hosts for later.
4. wait for drone to become queen.
5. queen lays eggs.
6. wait for eggs to hatch
7. facehugger people
8. wait for larvas to chestburst
9. wait for second larva generation to grow up
10. Oh wait, round end vote happened and escape shuttle arrives before you can do anything, crew didn't bother fighting you anyway and just went to escape. Yay quarantine? 

There is a long, _long_ windup time until xenos get going. By the time the first second generation hunter is created, the shuttle's already been called, since round extension votes never go through anyway. Now, you can argue this is also a balance question, and xenos can really get out of hand, so I am not interested in speeding up the xeno lifecycle in general.

However, I want xeno investations to not get totally trivialised by the round end all the time. One method to do that would have been to make the event fire earlier, but that has large knock-on effects I wished to avoid. Instead, I focused on one of the most boring parts of the xeno infestation:

Waiting for the initial larva to grow up. There really isn't anything to do during this part, either for the crew or the xenos. The xeno hide in vents, the crew is unaware of their presence. Now, in theory the larva could go out biting people to speed up their grow but that's basically suicide, and risking one of the two only xenos that are needed to get the infestation going is very unwise.

For that reason, the larvas are now created with 150/200 evolution progress. By the time they've found a good place to nest up, they should generally be ready to evolve now.  This means the action should start a few minutes sooner, meaning it has more of a chance to play out before shift end evacuation.

**Changelog:**
:cl:
tweak: midround event spawned xeno larva now need less time to grow up and evolve.
/:cl:

### CODE REVIEW

**Comment by Emanthealmighty** [comment]
Yes. No reason to make them wait so long. 
Every other event antag spawns in fully capable of doing things immediately, and so should the xenos.

**Comment by Dinarzad** [comment]
This also means Xenomorphs are going to steam roll that much harder because the growth time of larva was part of what is used to balance Xeno infestations.  The hive will be able to gush out fully capable hunters and sentinels in very little time,  before the crew even has a chance to really get a handle on the problem.

This isn't really a good thing in my eyes, if the crew are being ACTUALLY Diligent then they should by all means be rewarded for doing what they SHOULD do, this is similar to Blob. If all the crew respond to blobs immediately, blobs are usually handled relatively fast, but if ignored build into a ball of doom.
Xenos work on similar gameplay.  This change has greater balance repercussions then I feel is being given consideration.

I could see giving the INITIAL larva this treatment but any of the others should not be, the fact larva take so long to mature has even saved Xenos on more then one occasion as one or two get away and the long delay makes the crew complacent that they got them all while a new Queen forms somewhere else.

**Comment by TDSSS** [comment]
It's basically the same as if the event fired 2 minutes earlier. The speed at which more xenos are being produced stays the same.

**Comment by Quantum-M** [comment]
I think it should be halfway through instead of 3/4s of the way, because the survival of the larva can have a big outcome. If they die, the round continues as normal, if they survive and grow however, they can really change the direction of the round (and considering that in their current form xenos are a bit broken).

**Comment by taukausanake** [comment]
Xenomorph round really should call for quarantine as they are a dangerous alien lifeform much like the blob that can infect a whole station. If the shuttle was called the xenos could infect the shuttle and thereby CentComm when it lands

**Comment by Emanthealmighty** [comment]
> Xenomorph round really should call for quarantine as they are a dangerous alien lifeform much like the blob that can infect a whole station. If the shuttle was called the xenos could infect the shuttle and thereby CentComm when it lands

You've posted this on the wrong PR

**Comment by taukausanake** [comment]
> 10\. Yay quarantine?

**Comment by Couls** [comment]
It does call for quarantine but the crew transfer shuttle gets forced anyway

**Comment by davidchan13** [comment]
Honestly I'd be in favor of the Larva starting fully grown, can hand wave it that they were just hiding in the vents while waiting to mature before becoming sentient, since this only effects the two event spawned Larva it's hardly a balance concern. Could even argue for more larva to be spawned on high pop rounds.

**Comment by Ralta** [comment]
As long as this only occurs for the larva initially spawned by the event (which is what you appear to be doing), then I think this is a good change. It doesn't change much beyond allowing Xenos to start a little earlier than usual and being less likely to get face-stomped while waiting in larval form.

**Comment by Fox-McCloud** [comment]
We'll see how this goes, but Xenos need a lot of updates.
